sklearn/metrics/cluster/tests/test_unsupervised.py

```diff
@@ -48,7 +48,8 @@ def test_no_nan():
     D = np.random.RandomState(0).rand(len(labels), len(labels))
     silhouette = silhouette_score(D, labels, metric='precomputed')
     assert_false(np.isnan(silhouette))
-
+    silhouette_sample = silhouette_samples([[3],[3]], np.array([2,4]))
+    assert_false(np.isnan(silhouette_sample).any())
```

Comment:
This test is not a regression test. It will pass in master.

Can you remove this and add a new test testing something like this

    distances = np.random.RandomState(0).rand(5)
    A = _intra_cluster_distance(distance, np.array([0, 1, 1, 1, 1]), 0)
    assert_true(np.isnan(A))